src/core/classes/registry.ts

```diff
@@ -146,8 +152,8 @@ export default class Registry<T extends ComponentProto> extends EventTarget {
 				return null;
 			});
 
-		this.loading[id] = loadingComponent;
-		this.loadingList.push(loadingComponent);
+		this.loading[id] = loadingComponent as Promise<T>;
```

Comment:
We need a type assertion here since we excluded `null` from `Promise<T>` (it was `Promise<T | null>` before), but in the code, we still expect that some promises might reject with `null`.